extensions/wikia/Email/Controller/WallMessageController.class.php

```diff
@@ -160,14 +166,10 @@ class OwnWallMessageController extends WallMessageController {
 	 * @return string
 	 */
 	protected function getSummary() {
-		if ( $this->currentUser->isAnon() ) {
-			return $this->getMessage( 'emailext-wallmessage-anon-owned-summary' )
-				->parse();
-		} else {
-			return $this->getMessage( 'emailext-wallmessage-owned-summary',
-				$this->getCurrentProfilePage(), $this->authorUserName
-			)->parse();
-		}
+		return $this->getMessage( 'emailext-wallmessage-owned-summary',
+			\Title::newFromText( $this->getVal( 'threadId' ), NS_USER_WALL_MESSAGE )->getFullURL(),
```

Comment:
We should take care of this ```getVal``` up in ```initEmail```.  We can add a ```wallMessage``` (the Title  object created here) member variable to the ```WallMessageController``` base class  so that we can reuse that object down here in the sub classes.  We can also add a new assertion to ```assertMessageDetails``` for a valid ```wallMessage``` object.